src/daft-sql/src/planner.rs

```diff
@@ -1959,6 +1964,75 @@ impl<'a> SQLPlanner<'a> {
+                        let inner_dtype = sql_datatype(inner_dtype)?;
+                        let rest = rest
+                            .iter()
+                            .map(|p| {
+                                p.parse().map_err(|_| {
+                                    PlannerError::unsupported_sql(
+                                        "invalid tensor shape".to_string(),
+                                    )
+                                })
+                            })
+                            .collect::<SQLPlannerResult<Vec<_>>>()?;
+                        DataType::FixedShapeTensor(Box::new(inner_dtype), rest)
+                    }
+                },
+                "image" => match properties.as_slice() {
+                    [] => DataType::Image(None),
+                    [mode] => {
+                        let mode = mode.parse().map_err(|_| {
+                            PlannerError::unsupported_sql("invalid image mode".to_string())
+                        })?;
+                        DataType::Image(Some(mode))
+
+                    },
+                    [mode, height, width] => {
+                        let mode = mode.parse().map_err(|_| {
+                            PlannerError::unsupported_sql("invalid image mode".to_string())
+                        })?;
+                        let height = height.parse().map_err(|_| {
+                            PlannerError::unsupported_sql("invalid image height".to_string())
+                        })?;
+                        let width = width.parse().map_err(|_| {
+                            PlannerError::unsupported_sql("invalid image width".to_string())
+                        })?;
+                        DataType::FixedShapeImage(mode, height, width)
+                    }
+                    _ => invalid_operation_err!("invalid image properties"),
+                },
+                "embedding" => match properties.as_slice() {
+                    [inner_dtype, size] => {
+                        let inner_dtype = sql_datatype(inner_dtype)?;
+                        let Ok(size) = size.parse() else {
+                            invalid_operation_err!("invalid embedding size, expected an integer")
+                        };
+                        DataType::Embedding(Box::new(inner_dtype), size)
+                    }
+                    _ => invalid_operation_err!(
+                        "embedding must have datatype and size: ex: `embedding(int, 10)`"
+                    ),
+                },
+                "f64" => DataType::Float64,
+                "f32" => DataType::Float32,
+                "i64" => DataType::Int64,
+                "i32" => DataType::Int32,
+                "i16" => DataType::Int16,
+                "i8" => DataType::Int8,
+                "u64" => DataType::UInt64,
+                "u32" => DataType::UInt32,
+                "u16" => DataType::UInt16,
+                "u8" => DataType::UInt8,
```

Comment:
While we all love Rust (and perhaps some zig bias here too), I don't think we should expose these Rust type names on the SQL side since we already have many names for them. If we were coming from C, then this might read like so

```
"int64_t" => DataType::Int64
"int32_t" => DataType::Int32
```

```sql
CREATE TABLE T ( x int32_t );
```

This is quite odd for SQL, but perhaps feels natural to someone with a C background. Hopefully we can agree that, emphasized under a different context, bringing type names from language X into SQL is a little out-of-place. Even to me the INT32 is odd in SQL, but it's a lesser evil than INT4. The short names just seem so natural because we're Rust brained 😄 but not everyone is, especially those with SQL background.

Reply:
yeah I suppose we can revisit this later on. I don't see any harm in keeping the aliases, but I see where you are coming from.

Reply:
Sounds good. The only harm AFAIK is introducing more keywords than necessary in a language that's already bloated with keywords. Also the other type names should be reserved whereas these would be non-reserved which is maybe a bit odd to have two different keyword semantics on the same type. If we are really motivated to add these, then maybe a sqlparser-rs contrib is required to get them to be reserved keywords or maybe it's possible via some extensions mechanism. In summary, the fewer keywords the better especially reserved.